lib/backend/syncersv1/updateprocessors/globalnetworkpolicyprocessor_test.go

```diff
@@ -22,224 +22,151 @@ import (
 	"github.com/projectcalico/libcalico-go/lib/backend/model"
 	"github.com/projectcalico/libcalico-go/lib/backend/syncersv1/updateprocessors"
 	cnet "github.com/projectcalico/libcalico-go/lib/net"
-	"github.com/projectcalico/libcalico-go/lib/numorstring"
 )
 
 var _ = Describe("Test the GlobalNetworkPolicy update processor", func() {
-	name1 := "name1"
-	name2 := "name2"
-
-	v3GlobalNetworkPolicyKey1 := model.ResourceKey{
-		Kind: apiv3.KindGlobalNetworkPolicy,
-		Name: name1,
-	}
-	v3GlobalNetworkPolicyKey2 := model.ResourceKey{
-		Kind: apiv3.KindGlobalNetworkPolicy,
-		Name: name2,
-	}
-	v1GlobalNetworkPolicyKey1 := model.PolicyKey{
-		Name: name1,
-	}
-	v1GlobalNetworkPolicyKey2 := model.PolicyKey{
-		Name: name2,
-	}
-
-	It("should handle conversion of valid GlobalNetworkPolicys", func() {
+	emptyGNPKey := model.ResourceKey{Kind: apiv3.KindGlobalNetworkPolicy, Name: "empty"}
+	emptyGNP := apiv3.NewGlobalNetworkPolicy()
+
+	minimalGNPKey := model.ResourceKey{Kind: apiv3.KindGlobalNetworkPolicy, Name: "minimal"}
+	minimalGNP := apiv3.NewGlobalNetworkPolicy()
+	minimalGNP.Spec.PreDNAT = true
+	minimalGNP.Spec.ApplyOnForward = true
+
+	fullGNPKey := model.ResourceKey{Kind: apiv3.KindGlobalNetworkPolicy, Name: "simple"}
+	fullGNP := apiv3.NewGlobalNetworkPolicy()
+	fullGNP.Namespace = "default"
+	fullGNP.Spec.Order = &Order
+	fullGNP.Spec.Ingress = []apiv3.Rule{TestIngressRule}
+	fullGNP.Spec.Egress = []apiv3.Rule{TestEgressRule}
+	fullGNP.Spec.Selector = "mylabel == selectme"
+	fullGNP.Spec.DoNotTrack = true
+	fullGNP.Spec.PreDNAT = false
+	fullGNP.Spec.ApplyOnForward = true
+	fullGNP.Spec.Types = []apiv3.PolicyType{apiv3.PolicyTypeIngress}
```

Comment:
Should probably have type Egress as well, since there is an egress rule.

Reply:
Fixed